base/server/src/main/java/org/dogtagpki/server/cli/SubsystemCLI.java

```diff
@@ -0,0 +1,48 @@
+//
+// Copyright Red Hat, Inc.
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+//
+package org.dogtagpki.server.cli;
+
+import java.io.File;
+
+import org.apache.tomcat.util.net.jss.TomcatJSS;
+import org.dogtagpki.cli.CLI;
+import org.dogtagpki.cli.CommandCLI;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.netscape.cmscore.apps.CMS;
+import com.netscape.cmscore.apps.EngineConfig;
+import com.netscape.cmscore.base.ConfigStorage;
+import com.netscape.cmscore.base.FileConfigStore;
+
+/**
+ * @author Chris S. Kelley
+ */
+public abstract class SubsystemCLI extends CommandCLI {
+
+    public static final Logger logger = LoggerFactory.getLogger(SubsystemCLI.class);
+
+    protected SubsystemCLI(String name, String description, CLI parent) {
+        super(name, description, parent);
+    }
+
+    protected void initializeTomcatJSS() throws Exception {
+        TomcatJSS tomcatjss = TomcatJSS.getInstance();
+        tomcatjss.loadConfig();
+        tomcatjss.init();
+        logger.info("TomcatJSS inititialized");
+    }
+
+    protected EngineConfig getEngineConfig() throws Exception {
+        String catalinaBase = System.getProperty("catalina.base");
+        String subsystem = parent.getParent().getParent().getName();
```

Comment:
This would be a problem. The number of `getParent()` calls needed depends on the depth of the CLI. `pki-server <subsystem>-db-info` needs 2 but `pki-server <subsystem>-db-access-grant` needs 3.

Reply:
Ah, good spot. I shall fix.